lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/driver/evaluator/EvaluatorStatusManager.java

```diff
@@ -151,6 +172,10 @@ synchronized boolean isFailedOrKilled() {
     return EvaluatorState.FAILED == this.state || EvaluatorState.KILLED == this.state;
```

Comment:
I am not sure I like state check methods that have *"Or"* in their names. :-)
If we visualize state transition graph for the Evaluator, we'll see there are three levels:

   1. `ALLOCATED`, `SUBMITTED`, and `RUNNING` - let's call it **Available**, or Accessible, or Active, or something like that;
   2. `CLOSING`;
   3. `DONE`, `FAILED`, or `KILLED` - we can call it **Closed**, and the latter two can form a category **ClosedAbnormally**.

I think using that notation can make our code more readable and manageable.

**P.S.** Also, all *static* methods that check state really belong to the enumeration, e.g. `EvaluatorStatate` can have methods like
```java
  public final boolean isActive() {
    return this == ALLOCATED || this == SUBMITTED || this == RUNNING;
  }
```